### ISSUE TITLE
Crash Reports

### ISSUE THREAD

**Issue by Fargus5** [BalrogGames/herosfaith#15]
1) Crash Report: https://hastebin.com/owanuroxah.md
2) Crash Report https://hastebin.com/viyifumiqi.swift

**Comment by Fargus5** [comment]
Latest third crash report right now
[2017-12-28-10.log](https://github.com/BalrogGames/herosfaith/files/1590942/2017-12-28-10.log)

**Comment by Fargus5** [comment]
BumbleBee - Oggi alle 12:20
@Fargus Can you try running the server without spartan (and maybe largesk) and see if it crashes again

Fargus - Oggi alle 15:03
@BumbleBee ok testing it, I will tell you next days.

Fargus - Oggi alle 21:51
@BumbleBee  I removed spartan and large.sk as you told me and crashes are disappeared.
I will add spartan to understand better what is the reason and the cause of this problem. OK?

BumbleBee - Oggi alle 22:17
Ok

**Comment by Fargus5** [comment]
[11:58:05] [pool-6-thread-1/ERROR]: Could not pass event ChunkLoadEvent to Citizens v2.0.22-SNAPSHOT (build 1493)
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:499) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.Chunk.loadNearby(Chunk.java:910) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:53) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:137) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:120) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:116) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.getChunkAt(World.java:331) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.getChunkAtWorldCoords(World.java:327) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at ru.beykerykt.lightapi.server.nms.craftbukkit.CraftBukkit_v1_12_R1.sendChunkUpdate(CraftBukkit_v1_12_R1.java:97) [LightAPI.jar:?]
	at ru.beykerykt.lightapi.server.nms.craftbukkit.CraftBukkit_v1_12_R1.sendChunkUpdate(CraftBukkit_v1_12_R1.java:89) [LightAPI.jar:?]
	at ru.beykerykt.lightapi.request.RequestSteamMachine.run(RequestSteamMachine.java:109) [LightAPI.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: java.lang.IllegalStateException: Asynchronous entity add!
	at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:14) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.addEntity(World.java:1015) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.WorldServer.addEntity(WorldServer.java:1112) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.citizensnpcs.nms.v1_12_R1.util.NMSImpl.addEntityToWorld(NMSImpl.java:230) ~[?:?]
	at net.citizensnpcs.util.NMS.addEntityToWorld(NMS.java:44) ~[?:?]
	at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:199) ~[?:?]
	at net.citizensnpcs.EventListen.spawn(EventListen.java:580) ~[?:?]
	at net.citizensnpcs.EventListen.respawnAllFromCoord(EventListen.java:550) ~[?:?]
	at net.citizensnpcs.EventListen.onChunkLoad(EventListen.java:140) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor63.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_111]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	... 24 more
[11:58:05] [pool-6-thread-1/ERROR]: Could not pass event ChunkLoadEvent to Citizens v2.0.22-SNAPSHOT (build 1493)
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:499) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.Chunk.loadNearby(Chunk.java:910) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:53) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:137) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:120) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:116) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.getChunkAt(World.java:331) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.getChunkAtWorldCoords(World.java:327) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at ru.beykerykt.lightapi.server.nms.craftbukkit.CraftBukkit_v1_12_R1.sendChunkUpdate(CraftBukkit_v1_12_R1.java:97) [LightAPI.jar:?]
	at ru.beykerykt.lightapi.server.nms.craftbukkit.CraftBukkit_v1_12_R1.sendChunkUpdate(CraftBukkit_v1_12_R1.java:89) [LightAPI.jar:?]
	at ru.beykerykt.lightapi.request.RequestSteamMachine.run(RequestSteamMachine.java:109) [LightAPI.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: java.lang.IllegalStateException: Asynchronous entity add!
	at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:14) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.addEntity(World.java:1015) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.WorldServer.addEntity(WorldServer.java:1112) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.citizensnpcs.nms.v1_12_R1.util.NMSImpl.addEntityToWorld(NMSImpl.java:230) ~[?:?]
	at net.citizensnpcs.util.NMS.addEntityToWorld(NMS.java:44) ~[?:?]
	at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:199) ~[?:?]
	at net.citizensnpcs.EventListen.spawn(EventListen.java:580) ~[?:?]
	at net.citizensnpcs.EventListen.respawnAllFromCoord(EventListen.java:550) ~[?:?]
	at net.citizensnpcs.EventListen.onChunkLoad(EventListen.java:140) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor63.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_111]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	... 24 more
[11:58:05] [pool-6-thread-1/ERROR]: [Train_Carts] Error while initializing sign power states in chunk -5/24
java.lang.IllegalStateException: Asynchronous entity remove!
	at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:14) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.removeEntity(World.java:1143) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.WorldServer.j(WorldServer.java:1152) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.WorldServer.a(WorldServer.java:1123) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.Chunk.addEntities(Chunk.java:753) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:45) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:137) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:120) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:116) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.CraftWorld.getChunkAt(CraftWorld.java:131) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.CraftWorld.getBlockAt(CraftWorld.java:95) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.block.CraftBlock.getRelative(CraftBlock.java:193) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.block.CraftBlock.getRelative(CraftBlock.java:201) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.block.CraftBlock.getRelative(CraftBlock.java:197) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at com.bergerkiller.bukkit.tc.PowerState.get(PowerState.java:69) ~[TrainCarts.jar:?]
	at com.bergerkiller.bukkit.tc.PowerState.isSignPowered(PowerState.java:157) ~[TrainCarts.jar:?]
	at com.bergerkiller.bukkit.tc.PowerState.isSignPowered(PowerState.java:137) ~[TrainCarts.jar:?]
	at com.bergerkiller.bukkit.tc.RedstoneTracker.onChunkLoad(RedstoneTracker.java:106) [TrainCarts.jar:?]
	at sun.reflect.GeneratedMethodAccessor60.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_111]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:499) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.Chunk.loadNearby(Chunk.java:910) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:53) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:137) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:120) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:116) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.getChunkAt(World.java:331) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.getChunkAtWorldCoords(World.java:327) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at ru.beykerykt.lightapi.server.nms.craftbukkit.CraftBukkit_v1_12_R1.sendChunkUpdate(CraftBukkit_v1_12_R1.java:97) [LightAPI.jar:?]
	at ru.beykerykt.lightapi.server.nms.craftbukkit.CraftBukkit_v1_12_R1.sendChunkUpdate(CraftBukkit_v1_12_R1.java:89) [LightAPI.jar:?]
	at ru.beykerykt.lightapi.request.RequestSteamMachine.run(RequestSteamMachine.java:109) [LightAPI.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
[11:58:05] [pool-6-thread-1/ERROR]: Could not pass event ChunkLoadEvent to Citizens v2.0.22-SNAPSHOT (build 1493)
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:499) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.Chunk.loadNearby(Chunk.java:910) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:53) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:137) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:120) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:116) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.getChunkAt(World.java:331) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.getChunkAtWorldCoords(World.java:327) [spigot.jar:git-Spigot-8cbfc85-202da97]
	at ru.beykerykt.lightapi.server.nms.craftbukkit.CraftBukkit_v1_12_R1.sendChunkUpdate(CraftBukkit_v1_12_R1.java:97) [LightAPI.jar:?]
	at ru.beykerykt.lightapi.server.nms.craftbukkit.CraftBukkit_v1_12_R1.sendChunkUpdate(CraftBukkit_v1_12_R1.java:89) [LightAPI.jar:?]
	at ru.beykerykt.lightapi.request.RequestSteamMachine.run(RequestSteamMachine.java:109) [LightAPI.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: java.lang.IllegalStateException: Asynchronous entity add!
	at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:14) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.World.addEntity(World.java:1015) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.minecraft.server.v1_12_R1.WorldServer.addEntity(WorldServer.java:1112) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	at net.citizensnpcs.nms.v1_12_R1.util.NMSImpl.addEntityToWorld(NMSImpl.java:230) ~[?:?]
	at net.citizensnpcs.util.NMS.addEntityToWorld(NMS.java:44) ~[?:?]
	at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:199) ~[?:?]
	at net.citizensnpcs.EventListen.spawn(EventListen.java:580) ~[?:?]
	at net.citizensnpcs.EventListen.respawnAllFromCoord(EventListen.java:550) ~[?:?]
	at net.citizensnpcs.EventListen.onChunkLoad(EventListen.java:140) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor63.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_111]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot.jar:git-Spigot-8cbfc85-202da97]
	... 24 more
[11:58:06] [Server thread/INFO]: [Server] Server is restarting. World is being saved.

**Comment by Fargus5** [comment]
Timing report:
https://timings.spigotmc.org/?url=fudenofada

**Comment by Fargus5** [comment]
Hi


I want to inform you that after many days of testing and studying about this problem I finally understand the reason of the crash.


I need to agree with you saying that it is not spartan the cause of the crash but it was a memory leak problem of some plugins when Spartan is installed.


With the feature of “timings” I studied all the process of each plugins and I understand that the problem was the autorestart set on spigot.yml that was applied by the system when Spartan and these plugins run togheter.


I resolved it asking to the respective authors to fix this memory leak


The major one was “furniture lib”


Sorry for the late answer, but it take to me a long time to find, understand and solve it
